gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/task/DagTask.java

```diff
@@ -41,25 +42,33 @@
 public abstract class DagTask {
   @Getter public final DagActionStore.DagAction dagAction;
   private final MultiActiveLeaseArbiter.LeaseObtainedStatus leaseObtainedStatus;
+  private final DagActionStore dagActionStore;
   @Getter protected final DagManager.DagId dagId;
 
-  public DagTask(DagActionStore.DagAction dagAction, MultiActiveLeaseArbiter.LeaseObtainedStatus leaseObtainedStatus) {
+  public DagTask(DagActionStore.DagAction dagAction, MultiActiveLeaseArbiter.LeaseObtainedStatus leaseObtainedStatus,
+      DagActionStore dagActionStore) {
     this.dagAction = dagAction;
     this.leaseObtainedStatus = leaseObtainedStatus;
+    this.dagActionStore = dagActionStore;
     this.dagId = DagManagerUtils.generateDagId(dagAction.getFlowGroup(), dagAction.getFlowName(), dagAction.getFlowExecutionId());
   }
 
   public abstract <T> T host(DagTaskVisitor<T> visitor);
 
   /**
-   * Any cleanup work, e.g. releasing lease if it was acquired earlier, may be done in this method.
+   * Any cleanup work, including removing the dagAction from the dagActionStore and completing the lease acquired to
+   * work on this task, is done in this method.
    * Returns true if concluding dag task finished successfully otherwise false.
    */
   // todo call it from the right place
   public boolean conclude() {
     try {
+      // If dagAction is not present in store then it has been deleted by another participant, so we can skip deletion
+      if (this.dagActionStore.exists(this.dagAction)) {
```

Comment:
this is an unnecessary mysql call

Reply:
we should still handle this case for now we don't want the call to run into an exception

Reply:
for ex if we check that dagAction doesn't exist in store we call conclude directly and skip act then we want this check here. we can decide later.

Reply:
agreed: there's still a check-then-act race condition, so why not just make `deleteDagAction` idempotent, and avoid the pre-check?

Reply:
I guess delete is already idempotent by nature of where clause `"DELETE FROM %s WHERE flow_group = ? AND flow_name =? AND flow_execution_id = ? AND job_name = ? AND dag_action = ?";` no need to check then